lib/src/impl/platform_object.dart

```diff
@@ -8,18 +11,34 @@ import 'package:streams_channel/streams_channel.dart';
 /// where that live counterpart is held as a strong reference by the plugin
 /// implementation.
 abstract class PlatformObject {
-  final int _ablyHandle;
-  final AblyImplementation _ablyPlugin;
-  final int _handle;
 
+  int _handle;
 
-  PlatformObject(this._ablyHandle, this._ablyPlugin, this._handle);
+  PlatformObject(){
+    this.handle;  //fetching asynchronous handle proactively...
+  }
 
   @override
   String toString() => 'Ably Platform Object $_handle';
 
-  get handle => _handle;
-  get ablyHandle => _ablyHandle;
+  Future<int> createPlatformInstance();
+
+  bool _registering = false;
+  Future<int> get handle async {
+    if(_registering){
+      //if handle is queried while already being fetched from remote, it must be put on hold..
+      await Future.delayed(Duration(milliseconds: 250));
```

Comment:
@QuintinWillison improved the commentary fb7fa6459ff0dbafff2b180ce6474badc5925ed9
If this is still not clear, I can try giving it another shot.

Reply:
Thanks. Had a look at that.

If you re-read your commentary back I think there's a typo or too (e.g. 'server' > 'served').

Also, 250ms (4Hz) is relatively glacial when most apps should be rendering at a minimum of 60Hz (17ms frame to frame). I would suggest perhaps a delay of no more than 10ms would be more appropriate here. Apart from anything else that is more likely to drive out issues early on in testing too.

And, finally, should there not be a loop here checking `_registering` each time before assuming it's ok to proceed or do I misunderstand the intended outcome? And, if you do end up having a check loop, then there'll need to be some kind of escape route - e.g. timeout - in case the flag never clears. In fact your comment says "a short hold and re-check" but I'm not sure I see evidence of that in the code.

Reply:
"too" 🤦  😆

Reply:
"re-check" is happening via recursion. But as you pointed out, I feel it is appropriate to have a loop instead of recursion.
Also, I've added a timeout of 2 minutes with poll time reduced to 10ms from 250ms.

sha c792574efd37a3cff749aa066970262d0ff6dfec

Reply:
> Also, I've added a timeout of 2 minutes with poll time reduced to 10ms from 250ms.

Just checked your source... thankfully you've implemented "2 **seconds**". 😁

Reply:
Oops, my bad 😛

Reply:
The PR is merged already but I stumbled upon this code in another PR.

First I'd suggest to make the getter a function. 
In my opinion getters with side effects (except caching) are usually not a good idea (I admit this is a bit controversial) 
but after I saw this code
```dart
    this.handle;  //proactively acquiring handle
```
I wondered what the purpose of accessing a field might be. 
Then of course I thought this might be a getter.
Then I saw the comment and my bad feeling about this code got reinforced.

I'd suggest to make it a function named `acquireHandle()`

About the content. I think 

```dart
Future<int> acquireHandle() => _handle ??= createPlatformInstance()
    .timeout(const Duration(milliseconds: 2));
```
would do the same as the current code.

Is it OK to continue the discussion here or should I create an issue or should it be discussed in Slack?

Also it feels like either `createPlatformInstance` https://github.com/ably/ably-flutter/blob/0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/lib/src/impl/realtime/channels.dart#L34 
or `handle` should be `@protected` but I might not fully understand the purpose of the code yet.

Reply:
Not sure if this is related to this PR but perhaps the `reassemble` callback might be utilized if the handle needs to be reinitialized after hot-reload/restart 
https://proandroiddev.com/some-widely-used-override-methods-in-flutter-50696f08d42a#dc1c